From 8d4e0d40cf662aa39165a2d8f928ca7aa7a70b6c Mon Sep 17 00:00:00 2001
From: Simeon Marijon <simeon.marijon@foss.st.com>
Date: Tue, 20 Jun 2023 14:36:59 +0200
Subject: [PATCH 0445/1141] ARM: dts: stm32: add reboot mode based on NVMEM on
 STM32MP15

We use nvmem-cells to pass reboot mode to bootloader.
Normal, fastboot, recovery, serialx and usb boot modes are declared.
Boot mode value is stored in TAMP backup registers
(register 20 offset 0x50).

systemd command is need to set the boot mode.
e.g.
$ systemctl reboot ums_mmc0
Change-Id: If3a26b5bb14fc251e36cbf31c3afff10259e5ae5
Signed-off-by: Simeon Marijon <simeon.marijon@foss.st.com>
---
 arch/arm/boot/dts/stm32mp151.dtsi | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/arch/arm/boot/dts/stm32mp151.dtsi b/arch/arm/boot/dts/stm32mp151.dtsi
index c7e15d7c81a9..548268b5b2cf 100644
--- a/arch/arm/boot/dts/stm32mp151.dtsi
+++ b/arch/arm/boot/dts/stm32mp151.dtsi
@@ -2031,6 +2031,20 @@ m4_wakeup_area_hash: tamp-bkp@60 {
 					reg = <0x60 0x20>;
 				};
 			};
+
+			reboot_mode: reboot-mode {
+				compatible = "nvmem-reboot-mode";
+				nvmem-cells = <&boot_mode>;
+				nvmem-cell-names = "reboot-mode";
+				mode-normal = <0x00>;
+				mode-fastboot = <0x01>;
+				mode-recovery = <0x02>;
+				mode-stm32cubeprogrammer = <0x03>;
+				mode-ums_mmc0 = <0x10>;
+				mode-ums_mmc1 = <0x11>;
+				mode-ums_mmc2 = <0x12>;
+				mode-romcode_serial = <0xff>;
+			};
 		};
 
 		/*
-- 
2.39.2

